docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,158 @@
+
+Understanding through examples
+------------------------------
+
+Let us understand how to write tests for ``MyBear`` written for
+``python``.
+
+.. code::
+
+    from coalib.results.Result import Result
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class MyBear(LocalBear):
+
+        def run(file,filename):
+            if(len(file)>10):
+                yield Result(self, "Too many lines")
+
+**EXAMPLE 1** using ``verify_local_bear``
+
+.. code::
+
+    from bears.python.MyBear import MyBear
+    from tests.LocalBearTestHelper import verify_local_bear
+
+
+    good_file = '''
+
+    def HelloWorld():
+        print ('HelloWorld')
+
+    def main():
+        HelloWorld()
+
+    '''.splitlines(keepends=True)
+
+
+    bad_file = '''
+
+    def Hello():
+        return 'Hello'
+
+    def World():
+        return 'World'
+
+    def HelloWorld():
+        return Hello() + World()
+
+    def main():
+        HelloWorld()
+
+    '''.splitlines(keepends=True)
+
+    MyBearTest = verify_local_bear(MyBear,
+                                   valid_files=(good_file,),
+                                   invalid_files=(bad_file,))
+
+``good_file`` is a file which your bear considers as non-style-violating
+and a ``bad_file`` is one which has at least one error/warning/info.
+``My Bear`` checks if a file has less than or equal 10 lines. So you need to
+write a ``good_file`` which has less than or equal to 10 lines and a
+``bad_file`` which has more than 10 lines and feed them to
+``verify_local_bear`` as input along with your bear (MyBear in this case) and
+a few additional arguments.
+
+.. note::
+
+    ``good_file`` and ``bad_file`` are sequences just like ``file``. A ``file``
+    is a sequence of an input file.
+
+**EXAMPLE 2** using ``LocalBearTestHelper.check_validity``
+
+.. code::
+
+    from queue import Queue
+    from bears.python.MyBear import MyBear
+
+    from tests.LocalBearTestHelper import LocalBearTestHelper
+    from coalib.settings.Section import Section
+    from coalib.settings.Setting import Setting
+
+
+    class MyBearTest(LocalBearTestHelper):
+
+        def setUp(self):
+            self.section = Section('name')
+            self.section.append(Setting('max_number_of_lines', '10'))
+            self.uut = MyBear(self.section, Queue())
+
+        def test_valid(self):
+            self.check_validity(self.uut, ["import os"])
+
+        def test_invalid(self):
+            self.check_validity(self.uut, file, valid=False)
+
+.. note::
+
+    ``file`` contains more than 10 lines.
+
+``check_validity`` asserts if your bear yields any results for a particular
+check with a list of strings. First a *Section* and your Bear
+(in this case MyBear) is ``setUp``. Now your *Section* consists on default
+*Settings*. You can append any *Setting* depending on your test. Validate a
+check by passing your bear, lines to check as parameters
+(pass few others parameters if necessary) to ``check_validity``. The method
```

Comment:
1. it's other parameters, not others
2. i think it should be ``a few`` instead of ``few``, since those 2 have different meanings, and ``a few`` means ``some/ a small number``, which is the case.